pandas/core/generic.py

```diff
@@ -2458,8 +2462,22 @@ def to_msgpack(self, path_or_buf=None, encoding='utf-8', **kwargs):
         return packers.to_msgpack(path_or_buf, self, encoding=encoding,
                                   **kwargs)
 
-    def to_sql(self, name, con, schema=None, if_exists='fail', index=True,
-               index_label=None, chunksize=None, dtype=None, method=None):
+    # TODO: Replace `con: Any` and `Callable[[Any, Any, ...` when SQLTable
+    # and sqlalchemy can be imported. SQLTable can't be imported due to
+    # circular import. sqlalchemy can't be imported since it's an optional
+    # dependency.
+    def to_sql(self,
+               name: str,
+               con: Any,
+               schema: str = None,
+               if_exists: str = 'fail',
+               index: _bool = True,
+               index_label: Union[str, List[str]] = None,
+               chunksize: int = None,
+               dtype: Union[Dict[str, Dtype], Dtype] = None,
```

Comment:
Whoops sorry missed this but can you just import Dtype from `pandas._typing` and use that as the annotation here?

Reply:
You actually [mentioned](https://github.com/pandas-dev/pandas/pull/26795#discussion_r293600779) `Dict[Any, Dtype]` but I took the liberty to 'interpret' your comment, since I don't see how:
1.  `Dict[Any, Dtype]` annotates plain `Dtype` case and,
2. `Dict[str, Dtype]` is more precise for `dtype={'column_name': dtype}` case.

And now I don't see how  `dtype: Dtype` annotates `dtype={'column_name': Dtype}` situation. Seeing that you insist on this, I'll suppose I'm really off the mark here and change this as follows:

```
dtype: Dtype = None
```

After that I'll take another look at `core.dtypes` (and probably chase you down on Gitter soon).